tests/Unit/NumericalAlgorithms/Spectral/Test_LegendreGauss.cpp

```diff
@@ -0,0 +1,207 @@
+      Basis::Quadrature::Gauss
+    >(2);
+    for (size_t i = 0; i < 2; ++i) {
+      for (size_t j = 0; j < 2; ++j) {
+        CHECK(diff_matrix(i, j) == approx(diff_matrix_expected(i, j)));
+      }
+    }
+  }
+
+  SECTION("Check 3 points") {
+    const Matrix diff_matrix_expected = []() {
+      Matrix diff_matrix(3, 3);
+      diff_matrix(0, 0) = -1.936491673103709;
+      diff_matrix(0, 1) = 2.581988897471611;
+      diff_matrix(0, 2) = -0.645497224367902;
+      diff_matrix(1, 0) = -0.645497224367903;
+      diff_matrix(1, 1) = 0.;
+      diff_matrix(1, 2) = 0.645497224367903;
+      diff_matrix(2, 0) = 0.645497224367903;
+      diff_matrix(2, 1) = -2.581988897471611;
+      diff_matrix(2, 2) = 1.936491673103708;
+      return diff_matrix;
+    }();
+    const Matrix& diff_matrix = Basis::differentiation_matrix<
+      Basis::Polynomials::Legendre,
+      Basis::Quadrature::Gauss
+    >(3);
+    for (size_t i = 0; i < 3; ++i) {
+      for (size_t j = 0; j < 3; ++j) {
+        CHECK(diff_matrix(i, j) == approx(diff_matrix_expected(i, j)));
+      }
+    }
+  }
+
+}
+
+SPECTRE_TEST_CASE(
+    "Unit.Numerical.Spectral.LegendreGauss.LinearFilterMatrix",
+    "[NumericalAlgorithms][Spectral][Unit]") {
+  for (size_t n = 2; n < 10; ++n) {
+    const Matrix& filter_matrix = Basis::linear_filter_matrix<
+      Basis::Polynomials::Legendre,
+      Basis::Quadrature::Gauss
+    >(n);
+    const Matrix& grid_points_to_spectral_matrix =
+        Basis::grid_points_to_spectral_matrix<
+          Basis::Polynomials::Legendre,
+          Basis::Quadrature::Gauss
+        >(n);
+    const DataVector& collocation_points = Basis::collocation_points<
+      Basis::Polynomials::Legendre,
+      Basis::Quadrature::Gauss
+    >(n);
+    DataVector u(n);
+    for (size_t s = 0; s < n; ++s) {
+      u[s] = exp(collocation_points[s]);
+    }
+    DataVector u_filtered(n);
+    dgemv_('N', n, n, 1.0, filter_matrix.data(), n, u.data(), 1, 0.0,
+           u_filtered.data(), 1);
+    DataVector u_spectral(n);
+    dgemv_('N', n, n, 1.0, grid_points_to_spectral_matrix.data(), n,
+           u_filtered.data(), 1, 0.0, u_spectral.data(), 1);
+    for (size_t s = 2; s < n; ++s) {
+      CHECK(0.0 == approx(u_spectral[s]));
+    }
+  }
+}
+
+SPECTRE_TEST_CASE(
+    "Unit.Numerical.Spectral.LegendreGauss.InterpolationMatrix",
+    "[NumericalAlgorithms][Spectral][Unit]") {
+  auto check_interp = [](const size_t num_pts, auto func) {
+    const DataVector& collocation_points =
+        Basis::collocation_points<
+          Basis::Polynomials::Legendre,
+          Basis::Quadrature::Gauss
+        >(num_pts);
+    DataVector u(num_pts);
+    for (size_t i = 0; i < num_pts; ++i) {
+      u[i] = func(collocation_points[i]);
+    }
+    DataVector new_points{-0.5, -0.4837, 0.5, 0.9378, 1.0};
+    DataVector interpolated_u(new_points.size(), 0.0);
+
+    const Matrix interp_matrix =
+        Basis::interpolation_matrix<
+          Basis::Polynomials::Legendre,
```

Comment:
Consider type aliases

Reply:
Do you mean adding a type alias `LegendrePolynomials=Basis::Polynomials::Legendre` to `Basis.hpp`? I don’t quite see how that would make the code better - namespaces are more expressive, don’t you think?

Reply:
Or are you suggesting to add a typealias only to the scope in the test file?

Reply:
I think within the file is good (in cases where it is used a lot - where it is used only a single time I agree the full name is better)